lib/smack/SmackModuleGenerator.cpp

```diff
@@ -15,6 +15,11 @@ void SmackModuleGenerator::generateProgram(llvm::Module& m) {
 
   Naming naming;
   SmackRep rep(&getAnalysis<DSAAliasAnalysis>(), naming, program);
+  SmackBV* bv = &getAnalysis<SmackBV>();
+  if (SmackOptions::BitVectors || bv->isBVRequired())
```

Comment:
The `SmackBV` code looks like overkill. All we want to know is whether the current module uses particular instructions, and the `llvm::InstVisitor` class is made for that. You could extend `llvm::InstVisitor` to `BitvectorDetector` like below, and use it as follows:
````
BitvectorDetector bvd;
bvd.visit(m);
...
if (SmackOptions::BitVectors || bvd.usesBitvectors())
...
````
Here's a simple implementation of `BitvectorDetector`
````
class BitvectorDetector : public llvm::InstVisitor<BitvectorDetector> {
  bool found;
public:
  BitvectorDetector() : found(false) { }
  void visitInstruction(llvm::Instruction& I) {
    if (found)
      return;
    if (...) {
      found = true;
    }
  }
  void usesBitvectors() {
    return found;
  }
};
````